detector/calorimeter/dual-readout-tubes/src/DRTubesconstructor.cpp

```diff
@@ -763,22 +772,21 @@ void DRBarrelTubes::DRTubesconstructor::construct_calorimeter(Volume& calorimete
   Volume stave_volume("stave_volume", stave_solid, m_air);
   stave_volume.setVisAttributes(*m_description, "DRBTstave_vis");
 
-  // TowerID starts at 1, so that negative values can be used for the backward region
-  short int tower = 1;
-  // Place towers in theta direection into the stave as long we are in the barrel region
-  while (m_covered_theta < m_barrel_endcap_angle) {
-    std::cout << "----> DRBarrelTubes: tower = " << tower << std::endl;
-    Volume trap_volume("tower");
-    trap_volume.setMaterial(m_trap_material);
-
-    // Function in which the shape of the tower is calculated and constructed
-    this->construct_tower(trap_volume);
-
-    this->calculate_tower_position();
-    this->place_tower(stave_volume, trap_volume, tower);
-    this->increase_covered_theta(m_tower_theta);
-
-    tower++;
+    // Place towers in theta direection into the stave as long we are in the barrel region
+  for(int tower = 1; tower <= m_number_tower_end; tower++){
+    if (m_covered_theta < m_barrel_endcap_angle) {
+      this->increase_covered_theta(m_tower_theta);
+      if (tower < m_number_tower_start){
```

Comment:
I think you need to update this if statement, if I'm not mistaken?

```suggestion
      if (tower >= m_number_tower_start && tower<m_number_tower_end) {
```

Reply:
yes good spot thank you!